packages/keyring-controller/src/KeyringController.ts

```diff
@@ -2276,13 +2314,8 @@ export class KeyringController extends BaseController<
         throw new Error(KeyringControllerError.VaultDataError);
       }
 
-      await this.#restoreSerializedKeyrings(vault);
-
-      // The keyrings array and the keyringsMetadata array should
-      // always have the same length while the controller is unlocked.
-      if (this.#keyrings.length !== this.#keyringsMetadata.length) {
-        throw new Error(KeyringControllerError.KeyringMetadataLengthMismatch);
```

Comment:
Nit: We can delete this constant as well, since it's no longer used

Reply:
Done in [28afa46](https://github.com/MetaMask/core/pull/5725/commits/28afa4622f9c1fece73e8e0a25312504539b8605)